src/main/webapp/src/components/ratingList.js

```diff
@@ -31,71 +33,75 @@ function EnumeratedList(props) {
         {index + 1}
       </ListItemIcon>
       <ListItemText
-        primary={item.songName}
-        secondary={item.bandName + '. From album "' + item.albumName + '"'}
+        primary={item.name}
+        secondary={item.artist}
       />
     </ListItem>
   ));
   return <List>{reactListItems}</List>;
 }
 
 class RatingList extends React.Component {
+  constructor(props) {
+    super(props);
+
+    this.state = {
+      topSongs: [],
+      isLoading: false,
+      error: null,
+    };
+  }
+
+  componentDidMount() {
+    this.setState({ isLoading: true });
+
+    axios
+      .get("/top")
+      .then((result) => result.data)
+      .then((topSongs) =>
+        this.setState({
+          topSongs: topSongs,
+          isLoading: false,
+        })
+      )
+      .catch((error) =>
+        this.setState({
+          error,
+          isLoading: false,
+        })
+      );
+  }
+
   render() {
     const classes = this.props.classes;
+    const topSongs = this.state.topSongs;
+
+    if (this.state.error) {
+      console.log(this.state.error.message);
```

Comment:
In the future, it might good to think of a proper way to log messages on the server side